src/index.tsx

```diff
@@ -77,6 +77,7 @@ export type ResizeStartCallback = (
 ) => void | boolean;
 
 export interface ResizableProps {
+  as: any;
```

Comment:
I tried using `string | React.ComponentType` instead of `any` here but I faced a typing issue in the "ref" prop as a consequence and I was not able to solve it otherwise.

Reply:
@youknowriad Thanks for your work :)
Could you please specify HTMLElement name like following.

```
as: 'div' | 'header' | ...|  React.ComponentType<any>;
```

This is because we  need to ban elements that don't have children (i.e. `input`)

Reply:
If you are ok could you please story with `as`?

Reply:
Thanks for the review, I added a story and a testcase and replaced HTMLDivElement.
I've used `string | React.ComponentType<any>;` as a type. is that ok? The list of elements can be long :). 

Let me know what's missing thanks.